src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/services/ExceptionRecordToCaseTransformer.java

```diff
@@ -0,0 +1,114 @@
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.common.OcrFieldNames.EMAIL;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.common.OcrFieldNames.FIRST_NAME;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.common.OcrFieldNames.LAST_NAME;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.common.OcrFieldNames.LEGACY_ID;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.common.OcrFieldNames.POST_CODE;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.common.OcrFieldNames.POST_TOWN;
+
+@Service
+public class ExceptionRecordToCaseTransformer {
+
+    public static final String EVENT_ID = "createCase";
+    public static final String CASE_TYPE_ID = "Bulk_Scanned";
+
+    public SuccessfulTransformationResponse toCase(ExceptionRecord er) {
+        SampleCase caseData = buildCase(er);
+
+        return new SuccessfulTransformationResponse(
+            new CaseCreationDetails(CASE_TYPE_ID, EVENT_ID, caseData),
+            emptyList() // TODO: build warnings
+        );
+    }
+
+    private SampleCase buildCase(ExceptionRecord er) {
+
+        assertIsValid(er);
+
+        return new SampleCase(
+            getFromOcr(er, LEGACY_ID),
+            getFromOcr(er, FIRST_NAME),
+            getFromOcr(er, LAST_NAME),
+            getFromOcr(er, DATE_OF_BIRTH),
+            getFromOcr(er, CONTACT_NUMBER),
+            getFromOcr(er, EMAIL),
+            new Address(
+                getFromOcr(er, ADDRESS_LINE_1),
+                getFromOcr(er, ADDRESS_LINE_2),
+                getFromOcr(er, ADDRESS_LINE_3),
+                getFromOcr(er, POST_CODE),
+                getFromOcr(er, POST_TOWN),
+                getFromOcr(er, COUNTY),
+                getFromOcr(er, COUNTRY)
+            ),
+            er.scannedDocuments
+                .stream()
+                .map(it -> new ScannedDocument(
+                    it.type,
+                    it.subtype,
+                    it.url,
+                    it.controlNumber,
+                    it.fileName,
+                    it.scannedDate,
+                    it.deliveryDate,
+                    er.id
+                ))
+                .map(CcdCollectionElement::new)
+                .collect(toList())
+        );
+    }
+
+    private void assertIsValid(ExceptionRecord exceptionRecord) {
+
+        List<String> ocrFieldsInExceptionRecord =
+            exceptionRecord
+                .ocrDataFields
+                .stream()
+                .map(it -> it.name)
+                .collect(toList());
+
+        List<String> missingFields =
+            OcrFieldNames
+                .getRequiredFields()
+                .stream()
+                .filter(reqField -> !ocrFieldsInExceptionRecord.contains(reqField))
+                .collect(toList());
+
+
+        if (!missingFields.isEmpty()) {
+            throw new InvalidaExceptionRecordException(
```

Comment:
If you throw this then this message won't be displayed on CCD UI ?

Reply:
note this has nothing to do with http responses yet